src/main/java/model/data/EntityID.java

```diff
@@ -0,0 +1,58 @@
+package model.data;
+
+import model.JSONable;
+
+/**
+ * General class to unify
+ * 
+ * @author Wm. Keith van der Meulen
+ *
+ */
+public abstract class EntityID implements JSONable {
+	
+	protected String id;
```

Comment:
I think it is weird to have an id field, because that way all the subclasses store the information twice (and thus it is also twice in the JSON). Thus, I suggest to remove the field and just add an abstract getStringID method that needs to be implemented by all subclasses.